community/kernel/src/main/java/org/neo4j/kernel/impl/core/InMemoryTokenCache.java

```diff
@@ -59,18 +60,18 @@ public void putAll( List<TOKEN> tokens ) throws NonUniqueTokenException
 
         for ( TOKEN token : tokens )
         {
-            putAndEnsureUnique( newNameToId, token, tokenType );
             newIdToToken.put( token.id(), token );
+            putAndEnsureUnique( newNameToId, token, tokenType );
         }
 
-        nameToId.putAll( newNameToId );
         idToToken.putAll( newIdToToken );
+        nameToId.putAll( newNameToId );
     }
 
     public void put( TOKEN token ) throws NonUniqueTokenException
     {
-        putAndEnsureUnique( nameToId, token, tokenType );
         idToToken.put( token.id(), token );
+        putAndEnsureUnique( nameToId, token, tokenType );
```

Comment:
Why did you move the calls to `putAndEnsureUnique` after the puts to `idToToken`? If you didn't, you would not need to do the `remove` in `putAndEnsureUnique`.